app/src/main/java/swati4star/createpdf/fragment/ZipToPdfFragment.java

```diff
@@ -0,0 +1,158 @@
+package swati4star.createpdf.fragment;
+
+
+import android.app.Activity;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Environment;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.design.widget.Snackbar;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.dd.morphingbutton.MorphingButton;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+import swati4star.createpdf.R;
+import swati4star.createpdf.activity.MainActivity;
+import swati4star.createpdf.util.Constants;
+import swati4star.createpdf.util.FileUtils;
+import swati4star.createpdf.util.RealPathUtil;
+import swati4star.createpdf.util.StringUtils;
+
+import static android.app.Activity.RESULT_OK;
+
+public class ZipToPdfFragment extends Fragment {
+    private static final int INTENT_REQUEST_PICKFILE_CODE = 10;
+    private String mPath;
+    private Activity mActivity;
+
+    @BindView(R.id.selectFile)
+    MorphingButton selectFileButton;
+    @BindView(R.id.zip_to_pdf)
+    MorphingButton convertButton;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        View rootview = inflater.inflate(R.layout.fragment_zip_to_pdf, container, false);
+        ButterKnife.bind(this, rootview);
+        mActivity = getActivity();
+        return rootview;
+    }
+
+    @OnClick(R.id.selectFile)
+    public void showFileChooser() {
+        String folderPath = Environment.getExternalStorageDirectory() + "/";
+        Intent intent = new Intent();
+        intent.setAction(Intent.ACTION_GET_CONTENT);
+        Uri myUri = Uri.parse(folderPath);
+        intent.setDataAndType(myUri, "application/zip");
+
+        startActivityForResult(Intent.createChooser(intent, getString(R.string.merge_file_select)),
+                INTENT_REQUEST_PICKFILE_CODE);
+    }
+
+    public void onActivityResult(int requestCode, int resultCode, Intent data) throws NullPointerException {
+        if (data == null || resultCode != RESULT_OK || data.getData() == null)
+            return;
```

Comment:
Maybe we could show an error message to the user as well before returning.

Reply:
when starting chooser activity to select a zip file, we have used `intent.setDataAndType(myUri, "application/zip");`. so user will only be able to select zip files.
so above line 76 `return` will only execute if user has selected nothing. and this is not a error